File: kide-enterprise/backend/app/schemas/billing.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime
from app.models.subscription import PlanTier, SubscriptionStatus

class PlanResponse(BaseModel):
    tier: str
    name: str
    description: Optional[str] = None
    price_monthly_cents: int
    price_annual_cents: int
    max_projects: int # -1 = unlimited
    max_members: int  # -1 = unlimited
    monthly_ai_tokens: int # -1 = unlimited
    allowed_generators: List[str]
    traceability_export: bool
    audit_retention_days: int

class UsageSummaryResponse(BaseModel):
    projects_count: int
    max_projects: int
    members_count: int
    max_members: int
    ai_tokens_used: int
    monthly_ai_tokens: int

class SubscriptionResponse(BaseModel):
    id: int
    org_id: int
    plan_tier: str
    plan_name: str
    status: str
    current_period_start: Optional[datetime] = None
    current_period_end: Optional[datetime] = None
    trial_start: Optional[datetime] = None
    trial_end: Optional[datetime] = None
    is_trial: bool = False
    trial_days_remaining: int = 0
    cancel_at_period_end: bool = False
    usage: UsageSummaryResponse

class CreateCheckoutRequest(BaseModel):
    tier: PlanTier
    billing_cycle: str = "monthly" # monthly or annual
    success_url: Optional[str] = None
    cancel_url: Optional[str] = None

class CheckoutResponse(BaseModel):
    session_id: str
    checkout_url: str

class CreatePortalRequest(BaseModel):
    return_url: Optional[str] = None

class PortalResponse(BaseModel):
    portal_url: str

class ActivateTrialResponse(BaseModel):
    status: str
    plan_tier: str
    trial_end: datetime
    message: str

File: kide-enterprise/frontend/src/types/billing.ts
export type PlanTier = 'community' | 'team' | 'enterprise';

export interface Plan {
  tier: string;
  name: string;
  description?: string;
  price_monthly_cents: number;
  price_annual_cents: number;
  max_projects: number; // -1 for unlimited
  max_members: number;  // -1 for unlimited
  monthly_ai_tokens: number; // -1 for unlimited
  allowed_generators: string[];
  traceability_export: boolean;
  audit_retention_days: number;
}

export interface UsageSummary {
  projects_count: number;
  max_projects: number;
  members_count: number;
  max_members: number;
  ai_tokens_used: number;
  monthly_ai_tokens: number;
}

export interface Subscription {
  id: number;
  org_id: number;
  plan_tier: string;
  plan_name: string;
  status: string;
  current_period_start?: string;
  current_period_end?: string;
  trial_start?: string;
  trial_end?: string;
  is_trial: boolean;
  trial_days_remaining: number;
  cancel_at_period_end: boolean;
  usage: UsageSummary;
}

export interface CheckoutPayload {
  tier: PlanTier;
  billing_cycle?: 'monthly' | 'annual';
  success_url?: string;
  cancel_url?: string;
}

export interface CheckoutResponse {
  session_id: string;
  checkout_url: string;
}

export interface PortalResponse {
  portal_url: string;
}

export interface ActivateTrialResponse {
  status: string;
  plan_tier: string;
  trial_end: string;
  message: string;
}

